src/pages/UserDetails.js

```diff
@@ -0,0 +1,94 @@
+import { Fragment, useEffect, useState } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+import { ToastContainer, toast } from "react-toastify";
+import GoBack from "../common/buttons/Back";
+
+import { userDetailsRequest } from "../services/api";
+
+const UsersDetails = () => {
+  const navigate = useNavigate();
+  const { userId } = useParams();
+  const validUserId = parseInt(userId);
+  const [details, setDetails] = useState({});
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const navigateToPostsPage = () => {
+    return navigate(`/details/${validUserId}/posts`);
```

Comment:
use the routes constant here and use generatePath function to insert data into that string